docs/site/Model.md

```diff
@@ -465,6 +465,10 @@ The property decorator leverages LoopBack's
 [metadata package](https://github.com/strongloop/loopback-next/tree/master/packages/metadata)
 to determine the type of a particular property.
 
+{% include note.html content="
+Currently, property types must be specified explicitly either on the property itself or via the `type` option of the property decorator. Aliased types or types that extracted from a class or interface (e.g. `public name: OtherClass['otherProperty']`) will not work properly and will result in the property type being resolved as an empty object rather than the intended type in the generated OpenAPI specifcation. This is due to a limitation and flaw in the way TypeScript currently generates the metadata that is used to generate the OpenAPI specification for the application. (see [Issue #3863](https://github.com/strongloop/loopback-next/issues/3863))
+" %}
```

Comment:
Can you provide a code snippet that the `design:type` is not emitted by TypeScript to make it clear? For example:

```ts
@model()
class MyModel {
  @property()
  p1: string[]; // => design:type: Array
  p2: MyType<string>; => design:type Object
}
```

BTW is `public name: OtherClass['otherProperty']` even valid?

Reply:
@raymondfeng Thanks - I've added a snippet to the note in an amended commit.